mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -338,7 +363,16 @@ def invocations(data, content_type, model, input_schema):
             stack_trace=traceback.format_exc(),
         )
     result = StringIO()
-    predictions_to_json(raw_predictions, result)
+
+    # if the data was in an unsupported format, then we can assume that the user
+    # does not want the output to be formatted in the normal pyfunc way (otherwise
```

Comment:
What does the actual error look like for passing a dict input of previously unsupported structure to something like an sklearn model? Is it clear why it fails from within serving or is it just a generic error that is impossible to debug? Can we verify this behavior?

Reply:
+1 This is error-prone as we don't know what models customers are invoking against

Reply:
it looks like this—users can still see the error that made the underlying model fail:
<img width="901" alt="Screenshot 2023-12-20 at 4 04 25 PM" src="https://github.com/mlflow/mlflow/assets/148037680/421df1d7-3a7b-494f-8fa9-9975482bad4c">

Reply:
but yeah i do agree it's a little confusing. maybe we can add to the `message` something like 

"If you intended for your input to be converted to a Pandas Dataframe, please ensure your serialized input is a JSON object with exactly one of the following keys: {list of keys}. See {link to docs} for more information"

Reply:
I think we need more tests: For different flavor models, payload without supported keys, either works or fails with a clear error message. If it fails I think we should be able to instruct users to send a correct payload that used to work. 

Instead of this, I think it's easier if user's input doesn't contain any supported keys, we assume it's a valid dictionary and we add 'inputs' key for them then send it to the model. WDYT?

Reply:
To clarify, do you mean we add the "inputs" key and then continue parsing like normal? If so, wouldn't all the inputs get converted to `np.arrays`? I wonder if that makes it even more confusing for users.

Maybe another solution could be that we only enable this for custom pyfunc + transformers + langchain? If users are serving other types of models, then I agree this format would definitely not be helpful for them

Reply:
Currently `inputs` key works for both langchain & transformers models.

Reply:
I'm okay with this approach too, but what are the benefits of this over just passing the raw dict? If the input format is not compatible with the model, won't it still fail either way?

Reply:
Actually now that I think about it--if a user sends an input in this undefined format, it's pretty much guaranteed to fail on every flavor except custom pyfunc.

What do we think about restricting this feature to custom pyfunc models then? That way, other flavors can still retain their helpful failure messages.

Cc @dbczumar